Sources/ComposableArchitecture/Debugging/ReducerInstrumentation.swift

```diff
@@ -36,17 +36,20 @@ extension Reducer {
     let prefix = prefix.isEmpty ? zeroWidthSpace : "[\(prefix)] "
 
     return Self { state, action, environment in
+      var actionOutput: String!
       if log.signpostsEnabled {
-        os_signpost(.begin, log: log, name: "Action", "%s%s", prefix, debugCaseOutput(action))
+        actionOutput = debugCaseOutput(action)
+        os_signpost(.begin, log: log, name: "Action", "%s%s", prefix, actionOutput)
       }
       let effects = self.run(&state, action, environment)
       if log.signpostsEnabled {
         os_signpost(.end, log: log, name: "Action")
+        return
```

Comment:
We only wanna call `effectSignpost` in this `if`, right?

Reply:
yeah, that looks good